src/cmd/sign.js

```diff
@@ -111,38 +128,67 @@ export default function sign(
         log.warn('No extension ID specified (it will be auto-generated)');
       }
 
-      const signingResult = await signAddon({
+      if (useSubmissionApi && !channel) {
+        throw new UsageError(
+          'channel is a required parameter for the addon submission API'
+        );
+      }
+
+      if (useSubmissionApi && apiProxy) {
+        // https://github.com/mozilla/web-ext/issues/2472
+        throw new UsageError(
+          "apiProxy isn't yet supported for the addon submission API. " +
+          'See https://github.com/mozilla/web-ext/issues/2472'
+        );
+      }
+
+      try {
+        const apiHostURL = new URL(apiHost);
+        apiHost = apiHostURL.origin;
+      } catch (err) {
+        throw new UsageError(`Invalid apiHost: ${apiHost}`);
+      }
```

Comment:
In this version it seems we are going to validate the apiHost even when it is unused, not a big deal on its own given that its default value is valid.